pilot/cmd/pilot-agent/main.go

```diff
@@ -278,11 +285,20 @@ var (
 				log.Infof("Effective config: %s", out)
 			}
 
+			// check if SDS UDS path and token path exist, if both exist, requests key/cert
+			// using SDS instead of secret mount.
+			sdsEnabled := waitForFile(sdsUDSPath, time.Minute)
```

Comment:
Will this delay the start up of the pilot agent by an entire minute if not using SDS?

Reply:
I thought about this and agree this problem exist. 

The reason to wait is uds file is created at sds server start time, so there is a chance when pilot-agent start, the uds file doesn't exist. (though it usually takes much less time < 1min to be there).

one solution is adding another ```sdsenabled``` var variable explicitly(similar to original PR https://github.com/istio/istio/pull/13593), @costinm wdyt ?